src/tag.tsx

```diff
@@ -2,35 +2,95 @@ import * as React from 'react';
+    type?: 'promo' | 'active' | 'inactive' | 'success' | 'warning' | 'error';
     children: string;
-    color: string;
+    Icon?: React.FC<IconProps>;
+
+    /** @deprecated  use type prop */
+    color?: string;
 };
 
-const Tag: React.FC<TagProps> = ({children, color}) => {
-    const classes = useStyles({color});
+const Tag: React.FC<TagProps> = ({Icon, children, type = 'promo', color}) => {
+    const classes = useStyles({hasIcon: !!Icon});
     const {colors, isDarkMode} = useTheme();
-    // Hardcode black text in darkmode because there isn't a black text color constant that we can use in dark mode
-    const blackText = isDarkMode ? '#313235' : colors.textPrimary;
-    const textColor = color === colors.inverse ? blackText : colors.textPrimaryInverse;
+    const isInverse = useIsInverseVariant();
+
+    /**
+     * Legacy implementation
+     *
+     * @deprecated to be removed in the next major version
+     */
+
+    if (!children) {
+        return null;
+    }
+
+    if (color) {
+        // Hardcode black text in darkmode because there isn't a black text color constant that we can use in dark mode
+        const blackText = isDarkMode ? '#313235' : colors.textPrimary;
+
+        const textColor = color === colors.inverse ? blackText : colors.textPrimaryInverse;
+        return (
+            <ThemeVariant isInverse={false}>
+                <span className={classes.tag} style={{background: color}}>
+                    <Text color={textColor} size={14} lineHeight={20} weight="medium" truncate>
+                        {children}
+                    </Text>
+                </span>
+            </ThemeVariant>
+        );
+    }
+
+    const tagTypeToColors: Record<TagType, [string, string]> = {
+        promo: [colors.textTagPromo, colors.tagBackgroundPromo],
+        active: [colors.textTagActive, colors.tagBackgroundActive],
+        inactive: [colors.textTagInactive, colors.tagBackgroundInactive],
+        success: [colors.textTagSuccess, colors.tagBackgroundSuccess],
+        warning: [colors.textTagWarning, colors.tagBackgroundWarning],
+        error: [colors.textTagError, colors.tagBackgroundError],
+    };
+    const [textColor, backgroundColor] = tagTypeToColors[type];
+
+    const shouldUseInverseBackground = isInverse && !isDarkMode;
+
     return (
-        <span className={classes.tag}>
+        <span
+            className={classes.tag}
+            style={{background: shouldUseInverseBackground ? colors.inverse : backgroundColor}}
+        >
+            {Icon && (
+                <div style={{paddingRight: 4}}>
+                    <Icon color={textColor} size={pxToRem(16)} />
+                </div>
+            )}
             <ThemeVariant isInverse={false}>
-                <Text color={textColor} size={12} lineHeight={16} weight="regular" uppercase>
+                <Text color={textColor} size={14} lineHeight={20} weight="medium" truncate>
```

Comment:
is this what we want? can't we use a `Text2`? we don't want `Tags` to have bigger font size in desktop?

Reply:
Yes, this is documented in the spec:
![image](https://user-images.githubusercontent.com/579417/143438651-2828a732-0073-42c9-84d9-5176a98657a9.png)

Reply:
a comment in the code would be nice